slides/aml-25-042518-time-series/index.html

```diff
@@ -0,0 +1,952 @@
+```python
+ar25 = ar_model.AR(ppm[:500])
+res = ar25.fit(maxlag=25)
+ar_pred = res.predict(ppm.index[500], 
+                      ppm.index[-1])
+ppm[300:].plot(label="true")
+ar_pred.plot(label="AR predict")
+plt.legend(loc="best")
+```
+]
+.center[![:scale 90%](images/impact of order 2.png)]
+]
+
+???
+
+---
+
+# ARIMA
+
+.smaller[
+```python
+from statsmodels import tsa
+arima_model = tsa.arima_model.ARIMA (ppm[:500], order=(12, 1, 0))
+res = arima_model.fit()
+arima_pred = res.predict(ppm.index[500], ppm.index[-1], typ="levels")
+ppm[300:].plot(label="true")
+arima_pred.plot(label="ARIMA predict")
+plt.legend(loc="best")
+```
+]
+
+.left-column[
+
+.center[![:scale 90%](images/arima.png)]
+
+]
+
+.right-column[
+
+- More complicated process model
+- Internally uses differencing
+- Also checkout SARIMAX (state space model)
+
+]
+
+???
+
+---
+
+# 1d with sklearn
+
+.smaller[
+```python
+ppm.shape 
+```
+```
+(709,)
+```
+```python
+train = ppm[:500]
+test = ppm[500:]
+X = ppm.index.to_series().apply(lambda x: x.toordinal())
+X = pd.DataFrame(X)
+X_train, X_test = X.iloc[:500, :], X.iloc[500:, :]
+X_train.shape
+```
+```
+(500,1)
+```
+```python
+from sklearn.linear_model import LinearRegression
+lr = LinearRegression().fit(X_train, train)
+lr_pred = lr.predict(X_test)
+plt.plot(ppm.index, lr.predict(X))
+ppm.plot()
+```
+]
+
+
+???
+
+---
+
+# Linear model for trend
+
+.center[![:scale 85%](images/linear model for trend.png)]
+
+???
+
+---
+
+# Quadratic Model
+
+.smaller[
+```python
+from sklearn.preprocessing import PolynomialFeatures
+lr_poly = make_pipeline(PolynomialFeatures(include_bias=False), 
+                          LinearRegression())
+lr_poly.fit(X_train, train)
+plt.plot(ppm.index, lr_poly.predict(X))
+ppm.plot()
+```
+]
+
+.center[![:scale 55%](images/quadratic model.png)]
+
+
+???
+
+---
+
+# Detrending with trend model
+
+.smaller[
+```python
+y_res = ppm - lr_poly.predict(X)
+y_res.plot()
+```]
+
+.center[![:scale 60%](images/detrending with trend model.png)]
+
+???
+
+---
+class: compact
+
+# AR model on residuals
+
+.left-column[
+.smaller[
+```python
+from statsmodels.tsa import ar_model
+ar_model_res = ar_model.AR(y_res[:500])
+res_res = ar_model_res.fit(maxlag=12)
+ar_pred_res = res_res.predict(
+        ppm.index[500], ppm.index[-1])
+pred_ar_res = ar_pred_res + 
+              r_poly.predict(X_test)
+res_res.params
+```
+```
+const              -0.008039
+L1.interpolated     0.922508
+L2.interpolated    -0.243015
+L3.interpolated    -0.101173
+L4.interpolated    -0.077538
+L5.interpolated     0.158073
+L6.interpolated    -0.006164
+L7.interpolated    -0.023890
+L8.interpolated    -0.126381
+L9.interpolated     0.092664
+L10.interpolated   -0.025640
+L11.interpolated    0.414180
+L12.interpolated   -0.018078
+dtype: float64
+```
+]
+]
+
+.right-column[
+.center[![:scale 120%](images/ar model residuals.png)]
+
+]
+
+
+???
+FIXME - Note: Couldn't decrease size further
+
+
+---
+
+# Just Linear Regression
+
+.smaller[
```

Comment:
can you use tiny-code or smallest here to make it fit?

Reply:
done